### PR TITLE
Random tests

### PR DESCRIPTION
**Pull request by qbzzt**
1. Generate a random state test
2. Fill the test on `geth`
3. Run the filled test on `besu`
4. If it fails, copy to `/tmp` under a unique name
5. Go to step 1 (endless loop)

Not included yet (I'll do them when I have time): calls to precompiles.

### CODE REVIEW

File: src/Templates/RandomTests/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+   ```
+   docker run -p 8545:8545 -p 13001:30303 \
+      hyperledger/besu:develop retesteth --rpc-http-port 8545 \
+      --host-allowlist '*' # --logging ALL
+   ```
+
+1. Edit `config` to put the correct IP for besu and copy it to
+   `~/tests/config/besu/config`. You might need to do this as root.
+
+1. Run `./runTests.sh`. It is an endless loop that generates and runs
```
Comment (winsvega) [inline src/Templates/RandomTests/README.md:22]: can you make options for this script 
- testpath  
- docker / retesteth (when retesteth then original datadir by default is used)
- StateTest / BlockchainTest
- rounds <num> / unlimited (limit loop for testing)
- errored test output dir

File: src/Templates/RandomTests/oneTest.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+dir=`pwd`/../../..
+
+# Create a random test
+./randomTest.js > ../../GeneralStateTestsFiller/stExample/randomFiller.yml
+
+cd ~
+
+# Run the random test on Geth
+./dretesteth.sh -t $suite -- --testpath $dir --singletest $test \
+                --filltests --datadir /tests/config
```
Comment (winsvega) [inline src/Templates/RandomTests/oneTest.sh]: what if geth fails on test filling ?
should be reported too

**Review by winsvega — CHANGES_REQUESTED** [review]
add options to the script so I can run it on vps

File: src/Templates/RandomTests/oneTest.sh (hunk 2 above)
Comment (qbzzt) [reply src/Templates/RandomTests/oneTest.sh]: Added

File: src/Templates/RandomTests/README.md (hunk 1 above)
Comment (qbzzt) [reply src/Templates/RandomTests/README.md:22]: - [x]  testpath
- [x] docker / retesteth (when retesteth then original datadir by default is used)
- [x] StateTest / BlockchainTest
- [x] rounds / unlimited (limit loop for testing)
- [x] errored test output dir

Comment (qbzzt) [reply src/Templates/RandomTests/README.md:22]: I created all of those command line options. I still need to:

1. Update the README.md
2. Finish what I'm writing to have an explanation of the code.

**Comment by winsvega** [comment]
should keep in mind that besu require javahome to be set 
    `export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64`
for me it seems `JAVA_HOME` could not be set through `/etc/environment`
so I set it manually every time.

also example command: 
`./runTests.sh -n 1 -e /data/rndtests -r`

unfortunately on vps besu does not work (((

**Comment by qbzzt** [comment]
> should keep in mind that besu require javahome to be set `export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64` for me it seems `JAVA_HOME` could not be set through `/etc/environment` so I set it manually every time.
> 
> also example command: `./runTests.sh -n 1 -e /data/rndtests -r`
> 
> unfortunately on vps besu does not work (((

`oneTests.sh` doesn't run Besu, it connects to be an existing instance (specified in the retesteth config). So Besu can just as easily run in a VM outside the vps.

**Comment by winsvega** [comment]
can you name the test a little bit simplier. and the name inside of a test file should match the name of test file
because thats the rule we made for state tests

**Comment by qbzzt** [comment]
> can you name the test a little bit simplier. and the name inside of a test file should match the name of test file
> because thats the rule we made for state tests

Consider it done. The name is now <type of test>-<uuid>Filler.yml.

**Review by winsvega — APPROVED** [review]

**Pull request merged**